Detectors/Calibration/src/MeanVertexData.cxx

```diff
@@ -0,0 +1,114 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "DetectorsCalibration/MeanVertexData.h"
+#include "Framework/Logger.h"
+#include "MathUtils/MathBase.h"
+#include "CommonUtils/MemFileHelper.h"
+#include "CCDB/CcdbApi.h"
+#include "DetectorsCalibration/Utils.h"
+
+using namespace o2::calibration;
+
+namespace o2
+{
+namespace calibration
+{
+
+using Slot = o2::calibration::TimeSlot<o2::calibration::MeanVertexData>;
+using PVertex = o2::dataformats::PrimaryVertex;
+using o2::math_utils::math_base::fitGaus;
+using clbUtils = o2::calibration::Utils;
+
+//_____________________________________________
+MeanVertexData::MeanVertexData()
+{
+  LOG(INFO) << "Default c-tor, not to be used";
+}
+
+//_____________________________________________
+void MeanVertexData::print() const
+{
+  LOG(INFO) << entries << " entries";
+}
+
+//_____________________________________________
+void MeanVertexData::fill(const gsl::span<const PVertex> data)
+{
+  // fill container
+
+  LOG(INFO) << "input size = " << data.size();
+  for (int i = data.size(); i--;) {
+    // filling the histogram in binned mode
+    auto x = data[i].getX();
+    auto y = data[i].getY();
+    auto z = data[i].getZ();
+    LOG(INFO) << "i = " << i << " --> x = " << x << ", y = " << y << ", z = " << z;
+    uint32_t binx = (x + rangeX) * v2BinX;
+    uint32_t biny = (y + rangeY) * v2BinY;
+    uint32_t binz = (z + rangeZ) * v2BinZ;
```

Comment:
sorry for bad suggestion, actually, due to the rounding one has to do
````
auto dx = x+rangeX;
uint32_t binx = dx<0 ? 0xffffffff : (x + rangeX) * v2BinX;
...
````
BTW, I am preparing now flat histo class for VDrift calibration, can be eventually used also here.

Reply:
Thanks! I had copied and pasted blindly :-)